src/rustdoc-json-types/lib.rs

```diff
@@ -115,6 +115,32 @@ pub enum Visibility {
     },
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct DynTrait {
+    pub traits: Vec<PolyTrait>,
+    /// The lifetime of the whole dyn object
+    /// ```plain
+    /// dyn Debug + 'static
+    ///             ^^^^^^^
+    ///             |
+    ///             this part
+    /// ```
+    pub lifetime: Option<String>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct PolyTrait {
```

Comment:
I wasn't sure what the best name for this is. The compiler seems to use `PolyTrait` everywhere for this, but I don't think it's ever used outside that context.

Reply:
Do you know what it means ?

Reply:
A Trait with possible HRTB params

Reply:
Don't hesitate to add doc comments to explain what it stands for. Always appreciated.